python/ql/src/Functions/IterReturnsNonIterator.ql

```diff
@@ -12,17 +12,17 @@
 
 import python
 
-ClassObject return_type(FunctionObject f) {
+ClassValue return_type(FunctionValue f) {
     exists(ControlFlowNode n, Return ret |
-        ret.getScope() = f.getFunction() and
+        ret.getScope() = f.getScope() and
         ret.getValue() = n.getNode() and
-        n.refersTo(_, result, _)
+        result = n.pointsTo().getClass()
     )
 }
```

Comment:
Could all this be replaced with a call to `f.getAnInferredReturnType()`? (I would hope so)

Reply:
It seems to be so!